compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -107,6 +108,8 @@ pub struct SirFuncCx<'tcx> {
     pub func: ykpack::Body,
     /// Maps each MIR local to a SIR IPlace.
     var_map: FxHashMap<mir::Local, ykpack::IPlace>,
+    /// Maps TypeIds to their layout.
+    layout_map: FxHashMap<ykpack::TypeId, (usize, usize)>,
```

Comment:
How about `FxHashMap<ykpack::TypeId, ykpack::SizeAndAlign>`?

Reply:
Could do, but then would have to cast the values twice, once when putting it into the `layout_map`, and then again when taking them out, since `Layout` works on `usize`s.

Reply:
Fair enough. Don't worry then.